src/functions/tarsUpload/domain/__tests__/__mocks__/RecordingTARSUploader.ts

```diff
@@ -7,31 +7,52 @@ import { UploadRetryCount } from '../../upload/UploadRetryCount';
 export interface TARSUploaderCall {
   payload: ITARSPayload;
   interfaceType: TARSInterfaceType;
+  calledAt: number;
 }
 
 @injectable()
 export class RecordingTARSUploader implements ITARSUploader {
 
-  private errorForNextCall: Error | null = null;
+  private errors: Error[] = [];
   private calledWith: TARSUploaderCall[] = [];
   private retryCountForNextCall: number | null = null;
 
   uploadToTARS(payload: ITARSPayload, interfaceType: TARSInterfaceType): Promise<UploadRetryCount> {
-    this.calledWith = [...this.calledWith, { payload, interfaceType }];
-    if (this.errorForNextCall) {
-      return Promise.reject(this.errorForNextCall);
+    this.calledWith = [...this.calledWith, { payload, interfaceType, calledAt: Date.now() }];
+    if (this.errors.length > 0) {
+      const error = this.errors[0];
+      this.errors = this.errors.slice(1);
+      return Promise.reject(error);
     }
     return Promise.resolve(this.retryCountForNextCall || 0);
   }
 
   rejectWithErrorOnNextCall(error: Error) {
-    this.errorForNextCall = error;
+    this.errors = [error];
+  }
+
+  rejectWithErrorOnNextNCalls(error: Error, n: number = 1) {
+    // tslint:disable-next-line:prefer-array-literal
+    this.errors = [...this.errors, ...new Array(n).fill(error)];
```

Comment:
In what scenario would you want to add the error more that once?

As a side note, the variable `n` should have a more descriptive name.

Reply:
Improved the name. 👍 

You raise a good point, there's only one instance where I actually use this for anything other than 1.
```
    it('should throw the underlying error once retries have been exhausted', async () => {
      recordingUploader.rejectWithErrorOnNextNCalls(new TransientUploadError('502'), 3);
```

Could probably get by without this helper function, but not going to lose sleep over it.